src/main/scala/beam/taz/PointGenerator.scala

```diff
@@ -0,0 +1,43 @@
+package beam.taz
+
+import com.vividsolutions.jts.geom.{Geometry, GeometryFactory, PrecisionModel}
+import com.vividsolutions.jts.shape.random.RandomPointsInGridBuilder
+import org.matsim.api.core.v01.Coord
+
+import scala.annotation.tailrec
+
+trait PointGenerator {
+  def generate(geometry: Geometry, nPoints: Int): Seq[Coord]
+}
+
+object RandomPointsInGridGenerator extends PointGenerator {
+  private val projection: Int = 4326
+  private val geometryFactory: GeometryFactory = new GeometryFactory(new PrecisionModel(), projection)
+
+  override def generate(geometry: Geometry, nPoints: Int): Seq[Coord] = {
+    val pointsBuilder = new RandomPointsInGridBuilder(geometryFactory)
+    pointsBuilder.setExtent(geometry.getEnvelopeInternal)
+
+    @tailrec
+    def generate(n: Int): Seq[Coord] = {
+      pointsBuilder.setNumPoints(n)
+      val multiPoint = pointsBuilder.getGeometry.intersection(geometry)
+
```

Comment:
Have you tested the performance of this? As we are messing with large scales at times we need to be cognizant of that. I wonder if a view would be useful since you end up using a take in the longrun. No need to constantly run the code if we can put it off until the final point and reduce CPU cycles.